collect_app/src/main/java/org/odk/collect/android/activities/FillBlankFormActivity.kt

```diff
@@ -0,0 +1,351 @@
+import org.odk.collect.android.external.FormsContract
+import org.odk.collect.android.formmanagement.BlankFormListMenuDelegate
+import org.odk.collect.android.formmanagement.BlankFormsListViewModel
+import org.odk.collect.android.formmanagement.FormNavigator
+import org.odk.collect.android.injection.DaggerUtils
+import org.odk.collect.android.listeners.DiskSyncListener
+import org.odk.collect.android.network.NetworkStateProvider
+import org.odk.collect.android.preferences.dialogs.ServerAuthDialogFragment
+import org.odk.collect.android.projects.CurrentProjectProvider
+import org.odk.collect.android.tasks.FormSyncTask
+import org.odk.collect.android.utilities.ApplicationConstants
+import org.odk.collect.android.utilities.InstancesRepositoryProvider
+import org.odk.collect.android.utilities.MultiClickGuard.allowClick
+import org.odk.collect.android.views.ObviousProgressBar
+import org.odk.collect.androidshared.ui.DialogFragmentUtils.dismissDialog
+import org.odk.collect.androidshared.ui.DialogFragmentUtils.showIfNotShowing
+import org.odk.collect.geo.SelectItemFromMap
+import org.odk.collect.geo.SelectionMapActivity
+import org.odk.collect.permissions.PermissionListener
+import org.odk.collect.settings.keys.ProjectKeys
+import timber.log.Timber
+import javax.inject.Inject
+
+/**
+ * Responsible for displaying all the valid forms in the forms directory. Stores the path to
+ * selected form for use by [MainMenuActivity].
+ *
+ * @author Yaw Anokwa (yanokwa@gmail.com)
+ * @author Carl Hartung (carlhartung@gmail.com)
+ */
+class FillBlankFormActivity :
+    FormListActivity(),
+    DiskSyncListener,
+    OnItemClickListener,
+    LoaderManager.LoaderCallbacks<Cursor> {
+
+    @Inject
+    lateinit var networkStateProvider: NetworkStateProvider
+
+    @Inject
+    lateinit var blankFormsListViewModelFactory: BlankFormsListViewModel.Factory
+
+    @Inject
+    lateinit var currentProjectProvider: CurrentProjectProvider
+
+    @Inject
+    lateinit var instancesRepositoryProvider: InstancesRepositoryProvider
+
+    lateinit var selectFormFromMap: ActivityResultLauncher<Input>
+    lateinit var menuDelegate: BlankFormListMenuDelegate
+
+    private var formSyncTask: FormSyncTask? = null
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        DaggerUtils.getComponent(this).inject(this)
+
+        setContentView(R.layout.form_chooser_list)
+        title = getString(R.string.enter_data)
+
+        selectFormFromMap = registerForActivityResult(
+            SelectFormFromMap(),
+            EditInstanceResultCallback(
+                this,
+                FormNavigator(
+                    currentProjectProvider,
+                    settingsProvider,
+                    instancesRepositoryProvider::get
+                )
+            )
+        )
+
+        val blankFormsListViewModel = ViewModelProvider(this, blankFormsListViewModelFactory).get(
```

Comment:
Could be:
`val blankFormsListViewModel = ViewModelProvider(this, blankFormsListViewModelFactory)[BlankFormsListViewModel::class.java]`